alodataset/kitti_road.py

```diff
@@ -0,0 +1,173 @@
+        left_img_folder = os.path.join(self.split_folder, "image_2")
+        n_samples = len([file for file in os.listdir(left_img_folder) if file.startswith(f"{environement}_")])
+        self.extentions_dir = os.path.join(self.split_folder, self.get_split_folder())
+
+        self.items = {}
+        for idx in range(n_samples):
+            self.items[idx] = {
+                "left": os.path.join(self.extentions_dir, f"image_0/{environement}_{idx:06d}.png")
+                if grayscale
+                else os.path.join(self.split_folder, f"image_2/{environement}_{idx:06d}.png"),
+                "right": os.path.join(self.extentions_dir, f"image_{1 if grayscale else 3}/{idx:06d}.png")
+                if right_frame
+                else None,
+                "ground_truth": os.path.join(self.split_folder, f"gt_image_2/{environement}_{obj}_{idx:06d}.png")
+                if self.split == Split.TRAIN and grayscale is False
+                else None,
+                "calib": os.path.join(self.split_folder, f"calib/{environement}_{idx:06d}.txt"),
+            }
+
+    def __len__(self):
+        return len(self.items)
+
+    def getitem(self, idx) -> Dict[str, Frame]:
+
+        if self.sample:
+            return BaseDataset.__getitem__(self, idx)
+        item = self.items[idx]
+        calib = self._load_calib(item["calib"])
+        frames = {}
+
+        if item["right"] is not None:
+            frames["right"] = Frame(item["right"])
+            frames["right"].append_cam_extrinsic(CameraExtrinsic(calib[f"T_cam{1 if self.grayscale else 3}_rect"]))
+            frames["right"].append_cam_intrinsic(CameraIntrinsic(calib[f"K_cam{1 if self.grayscale else 3}"]))
+            frames["right"].baseline = calib["b_gray" if self.grayscale else "b_rgb"]
+
+        frames["left"] = Frame(item["left"])
+        frames["left"].append_cam_intrinsic(
+            CameraIntrinsic(np.c_[calib[f"K_cam{0 if self.grayscale else 2}"], np.zeros(3)])
+        )
+        frames["left"].append_cam_extrinsic(CameraExtrinsic(calib[f"T_cam{0 if self.grayscale else 2}_rect"]))
+        frames["left"].baseline = calib["b_gray" if self.grayscale else "b_rgb"]
+
+        if item["ground_truth"] is not None:
+            names = [self.obj, "valid_area", "invalid_area"]
+            color_codes = [[0, 255], [2, 255], [2, 0]]
+            # Cv2 loads images in BGR format
+            ground_truth = cv2.imread(item["ground_truth"], cv2.IMREAD_UNCHANGED)
+            for name, color_code in zip(names, color_codes):
+                area = ~np.array(ground_truth[:, :, color_code[0]] == color_code[1])
+                area = np.expand_dims(area, axis=0)
+                area = Mask(area, names=("C", "H", "W"))
+                frames["left"].append_mask(area, name=name)
+
+        return frames
+
+    # https://github.com/utiasSTARS/pykitti/tree/master
+    def read_calib_file(self, filepath):
```

Comment:
This function is repeated in all your classes with almost no changes, this is a good reason to create a parent class where you can add all repeated functions. Make other classes inherit from it to avoid functions redefinition. (with some small changes maybe)